server/src/routes/authorize.ts

```diff
@@ -58,6 +59,24 @@ router.post('/', async (req: Request, res: Response, next: Function) => {
             accountId: userInfo.account_id,
             displayName: userInfo.display_name,
         });
+
+        // if clientCode exists, create a separate plugin session
+        // first, check the user doc for the clientCode
+        const userDoc = await getUserByWebId(userInfo.account_id);
+        if (userDoc.clientCode !== clientCode) {
+            // looks like a different client/the UI initiated this login - don't create a plugin session
+            console.log(
+                '/authorize: User\'s clientCode does not match the OAuth state, no plugin session will be created',
+            );
+            return;
+        }
+
+        // remove clientCode from user
+        delete userDoc.clientCode;
+        await saveUser(userDoc);
+
+        // create a new plugin session including the clientCode
+        await createSession(userInfo, clientCode);
```

Comment:
Isn't it better to do this before sending the response? So the user does not get an invalid response in case any of this code fails